File: spec/javascript/test_expr_enum.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('ExprEnum', 'src/term_strz.bin', function(r, ExprEnum) {
  assert.equal(r.constDog, ExprEnum.Animal.DOG);
  assert.equal(r.derivedBoom, ExprEnum.Animal.BOOM);
  assert.equal(r.derivedDog, ExprEnum.Animal.DOG);
});

File: spec/python/test_expr_enum.py
import unittest

from expr_enum import ExprEnum

class TestExprEnum(unittest.TestCase):
    def test_expr_enum(self):
        r = ExprEnum.from_file("src/term_strz.bin")

        self.assertEqual(r.const_dog, ExprEnum.Animal.dog)
        self.assertEqual(r.derived_boom, ExprEnum.Animal.boom)
        self.assertEqual(r.derived_dog, ExprEnum.Animal.dog)
